### PR TITLE
Add ESLint loader for compile-time checks

### PR DESCRIPTION
**Pull request by nkbt**
Only for development.

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -52,11 +52,14 @@ module.exports = {
...
+    preLoaders: [
+      {test: /\.jsx?$/, loader: 'eslint', exclude: /build|node_modules/},
+    ],
   },
   resolve: {
     extensions: ['', '.js', '.jsx']
   },
-  plugins: plugins
+  plugins: plugins,
+  eslint: {configFile: '.eslintrc', emitWarning: true},
```
Comment (nkbt) [inline webpack.config.js]: Because of `emitWarnings: true` option build will not fail.

File: webpack.config.js [hunk 2]
```diff
@@ -22,8 +22,7 @@ if (DEV) {
   devtool = 'eval-source-map';
   loaders = ['react-hot'].concat(loaders);
   plugins = plugins.concat([
-    new webpack.HotModuleReplacementPlugin(),
-    new webpack.NoErrorsPlugin()
```
Comment (nkbt) [inline webpack.config.js]: Linter errors will not break the build. Actually having NoErrors for development does not make much sense anyway. Good for prod though.

**Comment by chenglou** [comment]
:+1: squash when you're done please

**Comment by nkbt** [comment]
Rebased and squashed.

**Pull request merged**